pkg/imagestore/remote/remote.go

```diff
@@ -0,0 +1,56 @@
+package remote
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/pivotal/image-relocation/pkg/image"
+	"github.com/pivotal/image-relocation/pkg/registry"
+
+	"github.com/deislabs/duffle/pkg/imagestore"
+)
+
+type Builder struct {
+	registryClient registry.Client
+}
+
+// remote is an image store which does not actually store images. It is used to represent thin bundles.
+type remote struct {
+	registryClient registry.Client
+}
+
+func NewRemoteBuilder() *Builder {
+	return &Builder{
+		registryClient: registry.NewRegistryClient(),
+	}
+}
+
+func (b *Builder) ArchiveDir(archiveDir string) imagestore.Builder {
+	return b
+}
+
+func (b *Builder) Logs(logs io.Writer) imagestore.Builder {
+	return b
+}
+
+func (b *Builder) Build() (imagestore.Store, error) {
+	return &remote{
+		registryClient: b.registryClient,
+	}, nil
+}
+
+func (r *remote) Add(im string) (string, error) {
+	return "", nil
```

Comment:
Yes, the `Add` method is called during `duffle export ... --thin` and it mustn't return an error otherwise exporting will fail.

The point of the image store abstraction is to allow both thin bundles (with an image store that keeps images in remote repositories) and thick bundles (with an image store that keeps images in an OCI image layout) to be handled by common code in exporting and relocation operations.

- `Add` is used during exporting. For a thick bundle, it stores an image in the thick bundle's OCI image layout. For a thin bundle, it doesn't need to do anything since the image is already available in a remote repository.
- The other method of the image store interface, `Push`, is used during relocation. For a thick bundle, it copies an image from the thick bundle's OCI image layout to the relocated repository. For a thin bundle, it copies an image from its original repository to the relocated repository.

(However, thanks for asking this question: I deleted some dead code as a result.)